packages/angular/projects/clr-angular/src/forms/common/control-container.ts

```diff
@@ -27,12 +26,7 @@ import { ClrControlSuccess } from '../common/success';
           shape="exclamation-circle"
           aria-hidden="true"
         ></clr-icon>
-        <clr-icon
-          *ngIf="showValid && controlSuccessComponent"
-          class="clr-validate-icon"
-          shape="check-circle"
-          aria-hidden="true"
-        ></clr-icon>
+        <clr-icon *ngIf="showValid" class="clr-validate-icon" shape="check-circle" aria-hidden="true"></clr-icon>
```

Comment:
Is there a test to make sure this appears only when showValid returns true and that it is not in the DOM when false?

Reply:
```
it('should have the success icon when valid', () => {
      container.state = CONTROL_STATE.VALID;
      fixture.detectChanges();
      const icon: HTMLElement = containerEl.querySelector('clr-icon[shape=check-circle]');
      expect(icon).toBeTruthy();
    });
```

There is this test, but you are right it will be better to update it a little more to be sure that this will be not visible when it's false